In this project you are trying out for an online marketplace company. The evaluation cover 3 important aspects of the daily job there that you need to know in order to get the position. 

The name of the database you will be working with is: data-analyst-web-store



Step 1:

Queries to know the data:

1. How many customers are there and what is the gender ratio?

2. How many items are there? Do you see any phenomena? (notice that item_id can be different)

3. What is the total amount of orders?

4. How many users were registered in July – December 2019 on the YouTube campaign?

5. How many users from the US bought the book Rich Dad? What is their share out of everyone who bought this book? And what is their share out of all US sales?

6. What is the most ordered item by quantity and what is the item with the highest total revenue? 

7. What is the share of total revenue, orders and users registration for each country, And how this ratio changes from 2019 to 2020 (show results for both years)

8. What is the total of shipments for each category in the data?

9. What is the average of the first order of each customer?

10. Does the membership split equally between gender? Do we get the same result per different countries in the data? Check only for English speaking countries (US, CA and EN)

11. What is the most successful campaign in terms of registrations? And what is the most successful campaign in terms of total revenue brought in?


In [1]:
import pandas as pd
from sqlalchemy import create_engine
import psycopg2

In [2]:
db_config = {'user': 'practicum_student',         # username             
                      'pwd': 's65BlTKV3faNIGhmvJVzOqhs', # password             
                      'host': 'rc1b-wcoijxj3yxfsf3fs.mdb.yandexcloud.net',       #host URL      
                      'port': 6432,              # connection port             
                      'db': 'data-analyst-web-store'}          # the name of the database

connection_string = 'postgresql://{}:{}@{}:{}/{}'.format(db_config['user'],
                                                                     db_config['pwd'],
                                                                       db_config['host'],
                                                                       db_config['port'],
                                                                       db_config['db'])

engine = create_engine(connection_string, connect_args={'sslmode':'require'})

In [3]:
def queryResult(q):
    return pd.io.sql.read_sql(q, con = engine)

In [4]:
query1 = '''
            SELECT gender,COUNT(DISTINCT customer_id)
            FROM customer_info
            GROUP BY gender
         '''
display(queryResult(query1))

gender  count
0  Female    241
1    Male    218



1. How many customers are there and what is the gender ratio?



In [5]:

query1 = '''
            select
                sum(case when "gender" = 'Male' then 1 else 0 end)/count(*)::float as male_ratio,
                sum(case when "gender" = 'Female' then 1 else 0 end)/count(*)::float as female_ratio,
                sum(customer_id) as total
            from 
                customer_info
         '''
display(queryResult(query1))

male_ratio  female_ratio   total
0    0.477083      0.522917  117859

2. How many items are there? Do you see any phenomena? (notice that item_id can be different)



In [17]:
query2 = '''
             select count(distinct item) as item_count, count(distinct item) as item_name_count  
             from items
         '''
display(queryResult(query2))

item_count  item_name_count
0         137              137

We can see that there's 175 unique items but there's only 137 unique name - The name can be reapeated and won't have the same price and id.

In [7]:
query1 = '''
            SELECT gender
            FROM customer_info
            GROUP BY gender
         '''
display(queryResult(query1))

gender
0  Female
1    Male

3. What is the total amount of orders?

In [8]:
query1 = '''
            
         '''
display(queryResult(query1))

ProgrammingError: (psycopg2.ProgrammingError) can't execute an empty query
[SQL: 
            
         ]
(Background on this error at: https://sqlalche.me/e/14/f405)

4. How many users were registered in July – December 2019 on the YouTube campaign?

In [ ]:
query1 = '''
            
         '''
display(queryResult(query1))

5. How many users from the US bought the book Rich Dad? What is their share out of everyone who bought this book? And what is their share out of all US sales?

In [ ]:
query1 = '''
            
         '''
display(queryResult(query1))

6. What is the most ordered item by quantity and what is the item with the highest total revenue? 

In [ ]:
query1 = '''
            
         '''
display(queryResult(query1))


7. What is the share of total revenue, orders and users registration for each country, And how this ratio changes from 2019 to 2020 (show results for both years)


In [ ]:
query1 = '''
            
         '''
display(queryResult(query1))


8. What is the total of shipments for each category in the data?


In [ ]:
query1 = '''
            
         '''
display(queryResult(query1))


9. What is the average of the first order of each customer?


In [ ]:
query1 = '''
            
         '''
display(queryResult(query1))


10. Does the membership split equally between gender? Do we get the same result per different countries in the data? Check only for English speaking countries (US, CA and EN)


In [ ]:
query1 = '''
            
         '''
display(queryResult(query1))


11. What is the most successful campaign in terms of registrations? And what is the most successful campaign in terms of total revenue brought in?


In [ ]:
query2 ='''

        '''